var/spack/repos/builtin/packages/gcc/detection_test.yaml

```diff
@@ -0,0 +1,46 @@
+paths:
+  # Ubuntu 18.04, system compilers without Fortran
+  - layout:
+      - subdir: [bin]
+        name: gcc
+        output: "echo 7.5.0"
```

Comment:
Can we make this take `"7.5.0"` and have the internal logic figure out how to make it print that output? `echo` isn't part of the output we care about at all.

Reply:
There are cases where we want to mock some specific behavior from an executable. For instance, a `gcc` that fails with an error message and an exitcode if called with `--version`, but returns some output if called with `-dumpversions`. To deal with that I renamed `output` to `script`, since that seems more appropriate given the nature of that field.